atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/Sweeper.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.sweep;
+
+import java.util.Set;
+import java.util.function.LongSupplier;
+
+import com.google.common.base.Optional;
```

Comment:
That would be great, but the caller of this class is in SweepTaskRunner which has the guava Optional imported and used over the place. And I need to save this optional in a local variable, so I'd need to either use a fully qualified class name or change all usages of Optional in SweepTaskRunner, which can snowball quite quickly. I've done that before and will not make this mistake again. If you want to switch to Java8 optionals everywhere, you should probably convert the whole project at once in a single commit. But that would cause pain in our internal projects.